src/main/java/Duke.java

```diff
@@ -81,34 +177,55 @@ public static void main(String[] args) {
         System.out.println();
         inWord = scan.nextLine();
 
-        //List of Special User Commands
-        String exitString = "bye";
-        String taskCommand = "list";
-        String doneCommand = "done";
 
         Task[] taskList = new Task[100];
         int index = 0;
+        String[] instruction = inWord.split("\\s+", 2);
+        String instructionType = instruction[0];
 
-        while (!inWord.toLowerCase().equals(exitString)) {
-            if (inWord.toLowerCase().equals(taskCommand)) {
+        while (!inWord.toLowerCase().equals(EXIT_STRING)) {
+            switch(instructionType) {
+            case LIST_COMMAND:
                 printList(Arrays.copyOf(taskList, index));
-                //Adding a Task
-            } else if (inWord.toLowerCase().startsWith(doneCommand)) {
-                String[] commands = inWord.split(" ");
-
-                //Check input validity (if user command starts with done and followed by an integer) and exceptions
-                if (commands.length != 2 || !isNumeric(commands[1])) {
+                break;
+            case DONE_COMMAND:
+                if (checkValidDoneInstruction(inWord)) {
+                    printTaskDone(inWord, index, taskList);
+                } else {
                     returnException();
+                }
+                break;
+            case EVENT_COMMAND:
+                if (checkValidEvent(inWord)) {
+                    printEvent(inWord, index, taskList);
+                    index++;
                 } else {
-                    printTaskDone(commands, index, taskList);
+                    returnException();
+                }
+                break;
+            case TODO_COMMAND:
+                if (checkValidTodo(inWord)) {
+                    printTodo(inWord, index, taskList);
+                    index++;
+                } else {
+                    returnException();
+                }
+                break;
+            case DEADLINE_COMMAND:
+                if (checkValidDeadline(inWord)) {
+                    printDeadline(inWord, index, taskList);
+                    index++;
+                } else {
+                    returnException();
                 }
-                //Mark Task as Done
-            } else {
-                addTask(index, inWord, taskList);
-                index += 1;
-                //Add additional task
+                break;
+            default:
+                returnException();
+                break;
             }
             inWord = scan.nextLine();
+            instruction = inWord.split("\\s+", 2);
+            instructionType = instruction[0];
```

Comment:
can add more comment to increase the readability.

Reply:
Thank you for the insightful feedback, I will address this issue in my next code update